### ISSUE TITLE
Tutorial for Sankey Diagram

### ISSUE THREAD

**Issue by NEPHILIM-1** [jonvitale/SenseColorableSankey#1]
If possible, please provide a tutorial on how to set up the data for a Sankey Diagram. It would be nice, for those of us who like to practice, if you could provide a sample data set and instructions on how to prep the data for the Sankey Diagram.

**Comment by jonvitale** [comment]
@NEPHILIM-1 

I've got a few things to do on this project - including a bug fix that a colleague found. Once that is done I will try to post something on the Qlik Community to describe this extension.

Jonathan

**Comment by Metti1371** [comment]
Hi, Your extension sounds promising based on what you described but when I want to insert it, for any dimension or measure I see this message: "Error: The calculation page is too large. Reduce the number of cells and try again". This cannot be right because I just tried with 3 columns with 9 rows. I would be great-full if you help me with this issue.

**Comment by jonvitale** [comment]
Metti, would it be possible for you to upload a data file that I could try to make this happen myself? Or, perhaps an inline table?

**Comment by Metti1371** [comment]
Thank you so much for reply. I have attached the excel . In sanki figure, I am trying to plot the years in x-axis with several columns (each column correspond to one year) and it shows the distribution of product. (in excel I was not sure how the column and rows should look like. I thought the years should be in each column. The layout is fake so if in your extension I have to distribute the data in different manner, please let me know.
[Test - Sanky.xlsx](https://github.com/jonvitale/SenseColorableSankey/files/1900506/Test.-.Sanky.xlsx)
)

**Comment by jonvitale** [comment]
Metti,

Thank you for bringing this to my attention. Honestly, at this moment, I have no idea what's going on, but I will investigate. I see the same error as you. The strange thing is that the original Sankey extension that I forked this repo from works fine with your data. Perhaps, for now, you want to use that one. 

https://github.com/xavierlp/SenseSankey

I'll come back to this thread when I figure out what's going on.

**Comment by jonvitale** [comment]
Metti,

I think I found the problem, I need to test on a couple different things, but I think it's a simple fix. At line 72 in the SenseColoarableSankey.js I am trying to call too many rows (qHeight): 10,000. That is the highest amount of allowable cells, not rows. So, I'm changing it to 1000. You can test it yourself and see if it works. Not sure, why it works for some data and not for others.

(if you do test it yourself, make sure to clear the cache after changing the code and also delete the original Sankey and make a new one).

**Comment by Metti1371** [comment]
Hi Jonathan,
You are geniuses. It is working now. I am just worried if I see the error for larger data. But thank you so much. This is literally the best Sanky that I have ever seen.

**Comment by Metti1371** [comment]
Hi Jonathan,
Thanks again for the quick fix. It was really useful. I was trying to play around with the image function in the sankey. I only could make it work when I set the dimensions to have the same image. I tried to set the measure but then It shows the same picture for all product and it only changes when I select one product. I attached the pictures for your consideration. I was wondering if you have any demo or tuturial or qvf file where you could set several pictures?

![1 without selection](https://user-images.githubusercontent.com/37508607/38735351-def01132-3f28-11e8-874d-c7743d4aefdc.PNG)
![2 after selection](https://user-images.githubusercontent.com/37508607/38735352-df08ed24-3f28-11e8-814a-e05cbb2ceead.PNG)

**Comment by jonvitale** [comment]
First of all, The Godfather - nice.

But, I'm sorry to say, that the image feature is not something I put into this extension, it was carried over from the original version that I forked from. In fact, I might take it out altogether because I can't really support it. So, I can't really help you at the moment on that one.

Do you really think you need it? If so, it might be something I consider working on later.

**Comment by Metti1371** [comment]
Dear Jonathan,
Thank you for quick reply. The godfather is one of my favorite so why not :)
Your extension is by far the best sankey I have ever seen in Qlik or other BI tool. The image function could be really useful (for me it is) since it gives a quick overview of what is happening in the flow. I do not want to push, but I would be grateful if you consider to test it. Thank you again for doing such a good job.

**Comment by ccastrilloc** [comment]
Hi, many thanks for the extension! I have a expression for dimension's color, but it hasn't any effect. The expression is something like if(Status='2',RGB(0,99,56),if(Status='1',RGB(255,128,0),RGB(127,127,0127)), the same for all the dimensions (4).
Hope you can help me. Many thanks in advance
Best regards

Cesar

**Comment by jonvitale** [comment]
Can you open a new issue, I'd like to be able to close them when they are resolved. This one is about tutorials. It also might help if you could upload an app.
